stl/src/syserror.cpp

```diff
@@ -113,29 +112,59 @@ _CRTIMP2_PURE unsigned long __CLRCALL_PURE_OR_CDECL _Winerror_message(
     unsigned long _Message_id, char* _Narrow, unsigned long _Size) { // convert to name of Windows error, return 0 for
                                                                      // failure, otherwise return number of chars
                                                                      // written pre: _Size < INT_MAX
-    const unique_ptr<wchar_t[]> _Wide(new wchar_t[_Size]); // not using make_unique because we want default-init
-    const auto _First = _Wide.get();
-    unsigned long _Wide_chars =
-        FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, 0, _Message_id, 0, _First, _Size, 0);
+    unsigned long _Chars = FormatMessageA(
+        FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, 0, _Message_id, 0, _Narrow, _Size, 0);
 
     // remove any trailing whitespace
-    while (_Wide_chars != 0) {
-        switch (_First[--_Wide_chars]) {
-        case L' ':
-        case L'\n':
-        case L'\r':
-        case L'\t':
-        case L'\0':
+    while (_Chars != 0) {

```

Comment:
This block is identical in the 2 functions. Extract?

Reply:
Good idea

Reply:
Resolved